### DISCUSSION TITLE
Unable to run minimal QuestPdf app on linux

### DISCUSSION THREAD

**Discussion by kenhales** [QuestPDF/QuestPDF#781] [Q&A] [answered]
I am trying to get a minimal program using QuestPDF to run in a linux docker container. It runs fine in Windows. I saw the "Going Production/Platform specific dependencies" page on the site (https://www.questpdf.com/going-production/platform-specific-dependencies.html) and included the two additional package references. Here is what I have:

QuestPdfContainerTest.csproj:
```
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>net8.0</TargetFramework>
    <ImplicitUsings>enable</ImplicitUsings>
    <Nullable>enable</Nullable>
  </PropertyGroup>

  <ItemGroup>
    <PackageReference Include="QuestPDF" Version="2023.12.5" />
    <PackageReference Include="SkiaSharp.NativeAssets.Linux.NoDependencies" Version="1.68.1" />
    <PackageReference Include="HarfBuzzSharp.NativeAssets.Linux" Version="7.3.0.1" />
  </ItemGroup>

</Project>
```

Program.cs:
```using QuestPDF.Fluent;

QuestPDF.Settings.License = QuestPDF.Infrastructure.LicenseType.Community;

Document
    .Create(container =>
    {
        container.Page(page =>
        {
            page.Header().Text("Hello PDF!");
        });
    })
    .GeneratePdf("Hello.pdf");

Console.WriteLine("Done.");
```

commands:
dotnet publish -c Release -o publish -r linux-x64
cd publish
dotnet QuestPdfContainerTest.dll

 I get this error:
Unhandled exception. System.TypeInitializationException: The type initializer for 'QuestPDF.Settings' threw an exception.
 ---> QuestPDF.Drawing.Exceptions.InitializationException: The QuestPDF library has encountered an issue while loading one of its dependencies. This type of error often occurs when the current runtime is missing necessary NuGet packages.

Please ensure the following NuGet packages are added to your primary/startup project and has matching versions:
- SkiaSharp.NativeAssets.Linux.NoDependencies
- HarfBuzzSharp.NativeAssets.Linux

For a detailed understanding, please examine the inner exception and consult the official SkiaSharp library documentation.
 ---> System.EntryPointNotFoundException: Unable to find an entry point named 'sk_compatpaint_new' in shared library 'libSkiaSharp'.

The error message says I need to include two packages, but the .csproj references both of them. The error message also says something about an entry point named 'sk_compatpaint_new', but I can't find any info about this error.

I have tried this on Ubuntu (Windows subsystem) and in a docker container running mcr.microsoft.com/dotnet/sdk:8.0 with the same results.

What am I missing?

**Comment by girlpunk** [comment]
Have you followed the instructions in the error message?

**Reply by kenhales** [reply]
Yes, please see my edited post, which shows the package references for SkiaSharp.NativeAssets.Linux.NoDependencies and HarfBuzzSharp.NativeAssets.Linux.

**Reply by MarcinZiabek** [reply, marked as answer]
Quite likely, the example code in the documentation shows outdated package versions. Please try installing both packages with the latest stable versions: 

```c#
SkiaSharp.NativeAssets.Linux.NoDependencies 2.88.7 // here you have 1.68.1
HarfBuzzSharp.NativeAssets.Linux 7.3.0.1
```

**Reply by kenhales** [reply]
That fixed it. Thank you.
